rust-spec/lightclient/detection/detection.md

```diff
@@ -96,148 +134,273 @@ there exist light blocks a(i) s.t.
+evaluates to true:
 
 - *b.Header.Height =  c.Header.Height* and
 - *sequ-rooted(b)* and
 - *sequ-rooted(c)*
 
 implies *b = c*.
 
+#### **[TMBC-BLOCKS-MCFORK.1]**
+
 If there exists two light blocks b and c, with *sign-unique(b,c) =
-false* then we have a *fork on the chain*.
+false* then we have a *fork*.
 
-> The following captures what I believe is called a light client fork
-> in our discussions. There is no fork on the chain up to the height
-> of block b. However, c is of that height, is different, and passes skipping
-> verification  
-> Observe that a light client fork is a special case of a slashable
-> fork.
+> The difference of the above definition to
+> [[TMBC-MC-FORK.1]](#TMBC-MC-FORK1) is subtle. The latter requires a
+> full node being affected by a bad block while
+> [[TMBC-BLOCKS-MCFORK.1]](#TMBC-BLOCKS-MCFORK1) just requires that a
+> bad block exists, possibly in memory of an attacker.
 
-#### **[TMBC-LC-FORK.1]**
+> The following captures a light client fork. There is no fork up to
+> the height of block b. However, c is of that height, is different,
+> and passes skipping verification. It is a stricter property than
+> [[TMBC-LC-ATTACK.1]](#TMBC-LC-ATTACK1), as
+> [[TMBC-LC-ATTACK.1]](#TMBC-LC-ATTACK1) requires that no correct full
+> node is affected.
+
+#### **[TMBC-BLOCKS-LCFORK.1]**
 
 Let *a*, *b*, *c*, be light blocks and *t* a time. We define
 *light-client-fork(a,b,c,t)* iff
 
 - *sign-skip-match(a,b,c,t) = false* and
 - *sequ-rooted(b)* and
 - *b* is "unique", that is, for all *d*,  *sequ-rooted(d)* and
-     *d.Header.Height=b.Header.Height* implies *d = b*
+     *d.Header.Height = b.Header.Height* implies *d = b*
 
-> Finally, let's also define bogus blocks that have no support.
-> Observe that bogus is even defined if there is a fork on the chain.
+> Finally, let us also define bogus blocks that have no support.
+> Observe that bogus is even defined if there is a fork.
 > Also, for the definition it would be sufficient to restrict *a* to
 > *a.height < b.height* (which is implied by the definitions which
-> unfold until *supports()*.
+> unfold until *supports()*).
 
 #### **[TMBC-BOGUS.1]**
 
 Let *b* be a light block and *t* a time. We define *bogus(b,t)* iff
 
 - *sequ-rooted(b) = false* and
-- for all *a*, *sequ-rooted(a)* implies *skip-root(a,b,t) = false*
+- for all *a*, *sequ-rooted(a)* implies *skip-trace(a,b,t) = false*
   
-> Relation to [fork accountability][accountability]: F1, F2, and F3
-> (equivocation, amnesia, back to the past) can lead to a fork on the
-> chain and to a light client fork.  
-> F4 and F5 (phantom validators, lunatic) cannot lead to a fork on the
-> chain but to a light client
-> fork if *t+1 < f < 2t+1*.  
-> F4 and F5 can also lead to bogus blocks
-
 ### Informal Problem statement
 
-> We put tags to informal problem statements as there is no sequential
-> specification.
+There is no sequential specification: the detector only makes sense
```

Comment:
There is a sequential specification using a global state, that contains a set of light blocks. You can use sing and support to define input output relations, i.e., a list, resp. a tree (see our discussion):).